src/client.js

```diff
@@ -174,6 +189,15 @@ class Client extends Connection {
     try {
       var des = this.deserializer.parsePacketBuffer(packet) // eslint-disable-line
     } catch (e) {
+      // Dump information about the packet only if user is not handling error event.
```

Comment:
this is messy, but not handling the `error` event will crash node process so I think we can get away with dumping some information for debugging here (previously we log gigantic hex dump to console).